### ISSUE TITLE
Search across files not always working without explicit folder being included

### ISSUE THREAD

**Issue by jblanchet** [microsoft/vscode#85969]
- VSCode Version: 1.40.2
- OS Version: 10.0.17763

Steps to Reproduce:
I don't know how many of these steps are necessary, but this is how I can reproduce it. 

1. With VS Code open at the root of a very large folder structure. No editors open.
2. Searching for: xboxconsole
3. files to include: *.build
4. Hit search, it will find a single instance. 
5. In the same situation, now set files to include: ./level1/level2/**/*.build
6. Multiple instances of 'xboxconsole' are in .build files underneath /level1/level2/level3. It now finds them all.

Does this issue occur when all extensions are disabled?: Yes

**Comment by roblourens** [comment]
Which OS, Windows? Wonder if it's hitting some windows path limit?  Can you give the absolute path to one of the files that's missing from the results?

**Comment by jblanchet** [comment]
Yes this is on Windows. Don't think its a path length issue, this is one of the full paths where the text resided but was not found:
D:\dev\na-test\TnT\Build\Icepick\plugins\Icepick.Unit\source\BenchmarkPlugin\BenchmarkPlugin.build
So in this case, I was searching from the root of the project, D:\dev\na-test\TnT, with the filtered search (looking for *.build), but didn't find it until I specifically searched with .\Build\Icepick\**\*.build.
The entire project (from D:\dev\na-test\TnT down) contains 275,000 files and 52,000 folders.

**Comment by roblourens** [comment]
I wonder if you could experiment with searching for files in other places to see if there is some depth where it stops working or some other pattern?

**Comment by roblourens** [comment]
Also check that it's not being excluded by a .gitignore file or setting, see https://github.com/microsoft/vscode/wiki/Search-Issues#missing-search-results. Searching with the path like that would override a gitignore file.

**Comment by jblanchet** [comment]
It was not being excluded by a .gitignore file or setting. 

I've tried to reproduce it now with other searches but don't seem able to, sorry.

**Comment by roblourens** [comment]
Let me know whether you see this issue again

**Comment by jblanchet** [comment]
Tried it this morning (I'm now on version 1.43.2).
My initial search now finds 55 matches, when I do the more specific search it finds 65 matches. So certainly better than it was when I posted this ticket, but still odd. 
I assume it has something to do with the size of the file set that I'm searching? In the first search, it's looking through 540k files. When I narrow it down to a specific subfolder, it's now looking through 232k files.
